app/src/main/java/com/android/voyageur/ui/trip/AddActivity.kt

```diff
@@ -84,12 +79,27 @@ fun AddActivityScreen(tripsViewModel: TripsViewModel, navigationActions: Navigat
 
     val today = normalizeToMidnight(Date())
     val dateNormalized = activityDate?.let { normalizeToMidnight(Date(it)) } ?: Date(0)
+    val selectedTrip = tripsViewModel.selectedTrip.value!!
+
+    if (activityDate != null &&
+        (dateNormalized.after(normalizeToMidnight(selectedTrip.endDate.toDate())) ||
+            dateNormalized.before(normalizeToMidnight(selectedTrip.startDate.toDate())))) {
+      Toast.makeText(
+              context, "Please select a date within the trip's scheduled dates", Toast.LENGTH_SHORT)
+          .show()
+      return
+    }
 
     if (activityDate != null && dateNormalized.before(today)) {
       Toast.makeText(context, "The activity date cannot be in the past", Toast.LENGTH_SHORT).show()
       return
     }
 
+    if (startTime == null && endTime != null) {
```

Comment:
Consider adding a validation to ensure that startTime is earlier than endTime, in order to prevent users from accidentally setting an invalid time range, which could lead to unexpected behavior or confusion.

Reply:
@BogdanCDev See https://github.com/swent-t16/Voyageur/blob/fix/activity-date-between-trip-dates/app/src/main/java/com/android/voyageur/ui/trip/AddActivity.kt#L157-L162